graphql_compiler/compiler/ir_lowering_gremlin.py

```diff
@@ -148,8 +148,9 @@ def validate(self):
             raise TypeError(u'Expected FoldScopeLocation fold_scope_location, got: {} {}'.format(
                 type(self.fold_scope_location), self.fold_scope_location))
 
+        allowed_block_types = (GremlinFoldedFilter, Traverse, Backtrack)
         for block in self.folded_ir_blocks:
-            if not isinstance(block, GremlinFoldedFilter):
+            if not isinstance(block, allowed_block_types):
                 raise AssertionError(u'Found non-GremlinFoldedFilter in folded_ir_blocks: {} {} '
```

Comment:
this error message is misleading now -- the allowed block types are more than just `GremlinFoldedFilter`